updatebeta

```diff
@@ -13,17 +13,39 @@ then
  MACH="armv7l"
 fi
 
+# Must be root to install
+
+if [[ $EUID -eq 0 ]]; then
+	echo "You are root."
+else
+	echo "Root access is needed, please run this command with sudo"
+	exit 1
+fi
 echo "Updating to latest beta version..."
+echo "Stop Domoticz and create backup of current install...
+# remove old backup
+rm -f domoticz.backup.tar.gz
+base=$(basename $PWD)
+cd ..
+# create backup in parent folder (mostly home)
+sudo service domoticz.sh stop
+tar --exclude $base/backups -czf $base.backup.tar.gz $base
+# move backup to domoticz folder
+mv $base.backup.tar.gz $base/$base.backup.tar.gz
+echo "Backup ready"
+cd $base
+echo "Downloading and installing new beta..."
 echo "Please Standby..."
 wget -4 -O domoticz_beta.tgz --no-check-certificate "https://www.domoticz.com/download.php?channel=beta&type=release&system=${OS}&machine=${MACH}"
 echo "Checking file Integrity..."
 tar -tzf domoticz_beta.tgz >/dev/null
 if [ $? -ne 0 ]
 then
  echo "Problem in downloaded Domoticz archive. Stopping update!"
+ sudo service domoticz.sh start
```

Comment:
This will leave you with a correct and partial Domoticz install.
Best to check if the wget command succeeded, if not, stop (and restart Domoticz)
You could check if the tar is correct, if not, stop (and restart Domoticz)
Extract , and restart Domoticz

But as mentioned, this script is also used internally and it will slow down the update proces quite a bit

Could you add a date/time timestamp in the filename as well ?